src/routes/transactionWrapper.js

```diff
@@ -10,17 +11,28 @@ const logContext = {
   namespace,
 };
 
-export default function transactionWrapper(originalFunction, context = '') {
+export default function transactionWrapper(originalFunction, context = '', isReadOnly = false) {
   return async function wrapper(req, res, next) {
     const startTime = Date.now();
     try {
       httpContext.set('auditDescriptor', originalFunction.name);
       // eslint-disable-next-line @typescript-eslint/return-await
       return await sequelize.transaction(async (transaction) => {
         httpContext.set('transactionId', transaction.id);
+        let snapShot;
         try {
           await addAuditTransactionSettings(sequelize, null, null, 'transaction', originalFunction.name);
+
+          if (isReadOnly) {
+            snapShot = await captureSnapshot(true);
+          }
+
           const result = await originalFunction(req, res, next);
+
+          if (isReadOnly && await hasModifiedData(snapShot, transaction.id)) {
+            throw new Error('Transaction was flagged as READONLY, but has modifed data.');
```

Comment:
The txn would be rolled back correct? Maybe we should mention that in the exception.